src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -17,28 +24,172 @@ class GaussianIVPFilter(ODESolver):
+
+        .. math:: g(m) = H_1 m(t) - f(t, H_0 m(t))
+
+        and user-specified measurement noise covariance :math:`R`. Almost always, the measurement noise covariance is zero.
+
+        Compute either type filter, each with a different interpretation of the Jacobian :math:`J_g`:
+
+        - EKF0 thinks :math:`J_g(m) = H_1`
+        - EKF1 thinks :math:`J_g(m) = H_1 - J_f(t, H_0 m(t)) H_0^\\top`
+        - UKF thinks: ''What is a Jacobian?'' and uses the unscented transform to compute a tractable approximation of the transition densities.
+        """
+        measurement_model_string = measurement_model_string.upper()
+
+        # This dict will be made much smaller once "probsolve_ivp" is leaned out.
+        choose_meas_model = {
+            "EKF0": pnfs.DiscreteEKFComponent.from_ode(
+                ivp,
+                prior=prior,
+                ek0_or_ek1=0,
+                evlvar=measurement_noise_covariance,
+                forward_implementation="sqrt",
+                backward_implementation="sqrt",
+            ),
+            "EKS0": pnfs.DiscreteEKFComponent.from_ode(
+                ivp,
+                prior=prior,
+                ek0_or_ek1=0,
+                evlvar=measurement_noise_covariance,
+                forward_implementation="sqrt",
+                backward_implementation="sqrt",
+            ),
+            "EKF1": pnfs.DiscreteEKFComponent.from_ode(
+                ivp,
+                prior=prior,
+                ek0_or_ek1=1,
+                evlvar=measurement_noise_covariance,
+                forward_implementation="sqrt",
+                backward_implementation="sqrt",
+            ),
+            "EKS1": pnfs.DiscreteEKFComponent.from_ode(
+                ivp,
+                prior=prior,
+                ek0_or_ek1=1,
+                evlvar=measurement_noise_covariance,
+                forward_implementation="sqrt",
+                backward_implementation="sqrt",
+            ),
+            "UKF": pnfs.DiscreteUKFComponent.from_ode(
+                ivp,
+                prior,
+                evlvar=measurement_noise_covariance,
+            ),
+            "UKS": pnfs.DiscreteUKFComponent.from_ode(
+                ivp,
+                prior,
+                evlvar=measurement_noise_covariance,
+            ),
+        }
+
+        if measurement_model_string not in choose_meas_model.keys():
+            raise ValueError("Type of measurement model not supported.")
+
+        return choose_meas_model[measurement_model_string]
+
+    # Construct an ODE solver from different initialisation methods.
+    # The reason for implementing these via classmethods is that different
+    # initialisation methods require different parameters.
+
+    @classmethod
+    def from_rk_init(
+        cls,
+        ivp,
+        prior,
+        measurement_model,
+        with_smoothing,
+        init_h0=0.01,
+        init_method="DOP853",
+    ):
```

Comment:
I personally found it cleaner to choose initialisation with these classmethods, while demanding the abstract interface (a function that implements an initialisation) in the __init__. The reason is mostly that `from_rk_init` has a few special parameters, which are not relevant for the other method(s). and  that this way it will be easier to try a new implementation method by writing the function and "plugging it into the ODE solver" without touching the source code.

Reply:
I like the modularity of passing functions :+1: what about `with_rk_init` and `with_taylormode_init`? Personally, that's a bit closer to how I would describe what's happening. On the other hand, "from" seems to be a thing with classmethods. So, no strong opinion, use whichever you prefer :)

Reply:
I see what you're saying here, but AFAIK `from*` is what is used with class methods (everyone that sees `Bla.from_xyz` gets the idea that it is a creation method.